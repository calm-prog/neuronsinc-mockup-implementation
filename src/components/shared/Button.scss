@use '../shared/variables' as *;

.button {
  width: 12rem;
  text-align: center;
  background: linear-gradient(89.62deg, #423d55 0.32%, #7a7390 99.67%);
  border-radius: 10px;
  color: #fff;
  margin-left: auto;
  margin-right: auto;
  padding: 0.7rem 0;
  font-size: 0.8rem;
  border: none;
  display: block;
  font-family: "Montserrat", sans-serif;
  transition: 0.1s;
  box-shadow: 0 0 0rem 0rem $input-primary;
  cursor: pointer;

  span {
    display: inline-block;
    transform: translateX(0);
    transition: 0.3s;
    padding: 0.15rem;
    border-radius: 0.3rem;

    &:after {
      content: "\203A";
      position: absolute;
      opacity: 0;
      bottom: 0.67rem;
      transform: translateX(1rem);
      transition: 0.3s;
      height: 1rem;
      width: 1rem;
      border-radius: 100%;
      font-size: 1.5rem;
    }
  }

  &:hover span {
    transform: translateX(-0.5rem);

    &:after {
      opacity: 1;
      transform: translateX(0.7rem);
    }
  }

  &:focus {
    outline: none;
    box-shadow: 0 0 0rem 0.15rem $input-primary;
  }
}

@media (max-width: 768px) {
    .button {
        width: 100%;
    }
}
