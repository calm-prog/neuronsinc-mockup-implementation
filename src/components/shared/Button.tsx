import './Button.scss'

interface Props {
    text: string
    cssClass?: string
}

const Button: React.FC<Props> = ({text, cssClass=""}) => {
    return (
        <button className={"button" + cssClass} data-testid="button">
            <span>{text}</span>
        </button>
    )
}   

export default Button