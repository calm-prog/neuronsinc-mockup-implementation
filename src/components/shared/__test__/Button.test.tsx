import renderer from 'react-test-renderer'
import { render, screen, cleanup } from '@testing-library/react'
import Button from '../Button'

afterEach(() => {
    cleanup();
})

test("renders button text", () => {
    const component = renderer.create(<Button text="test btn"></Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test("should render Button component", () => {
    render(<Button text="test btn"></Button>);
    const buttonElement = screen.getByTestId("button");
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent("test btn");
})