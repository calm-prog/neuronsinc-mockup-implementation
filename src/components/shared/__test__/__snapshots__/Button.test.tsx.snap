// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button text 1`] = `
<button
  className="button"
  data-testid="button"
>
  <span>
    test btn
  </span>
</button>
`;
